### DISCUSSION TITLE
showError not working

### DISCUSSION THREAD

**Discussion by misochenk0** [nuxt/nuxt#17012] [Questions]
Hi
I'm trying to upload localisation depending on route params, and if any error comes to catch block I want to redirect on 404 page, but instead got this error and no redirect at all.
![image](https://user-images.githubusercontent.com/78084319/186449371-e4ef9087-5e59-407a-9006-29e04355a775.png)
 
 How i can fix this or maybe i done something wrong? 
 thanks for any help (nuxt-3 rc-8)
![image](https://user-images.githubusercontent.com/78084319/186448731-0c1e0a75-46b2-4c1b-989d-0f6b93df60ba.png)

*404 page working as expected if i already opened it by entering not existing route manually after that code works correctly. But if 404 wasnt opened later it still throws error "cannot read properties"

**Comment by danielroe** [comment]
You should use `throw createError` instead. `showError` is not meant to be thrown.

The issue, however, might be related to https://github.com/nuxt/framework/issues/6899. Do you have a reproduction?

update - you probably shouldn't throw error just before mount either. can you move it to `onMounted`?

**Reply by misochenk0** [reply]
Hi, I found out that the error occurs when using a teleport with a "to" attribute and a selector specified, and I was also able to reproduce that error
https://stackblitz.com/edit/nuxt-starter-4pccrt?file=pages%2Findex.vue

**Comment by misochenk0** [comment]
Problem was solved by adding v-if on <teleport> with variable loaded, which becomes true after component mounted hook triggers. 
Odd but it works  
`<teleport v-if="loaded" to=".div">`

`data() {
        return {
            loaded: false,
        }
  },
  mounted() {
      this.loaded = true
 },
`
